src/components/poolstats/PoolPositionWidget.tsx

```diff
@@ -39,10 +49,175 @@ const PerformanceCard = styled.div`
+      const getPrices03M = axios.get(`${API_URL}/token_returns/${poolData.token0Address}/1/3m/${endTime.toFixed(0)}`);
+      const getPrices13M = axios.get(`${API_URL}/token_returns/${poolData.token1Address}/1/3m/${endTime.toFixed(0)}`);
+
+      axios.all([getEvents, getPoolReturns, getPrices0, getPrices1, getPoolReturns3M, getPrices03M, getPrices13M]).then(
+        axios.spread((eventData, poolReturnsData, price0Data, price1Data, poolReturnsData3M, price0Data3M, price1Data3M) => {
+          const events = eventData.data.result;
+          const transfers = events.filter((ev: any) => ev.topics[0] === '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef');
+
+          let poolReturns = (poolReturnsData.data as PoolReturns).concat(...(poolReturnsData3M.data as PoolReturns));
+          poolReturns = poolReturns.filter(r => r.total_supply !== null);
+          const prices0 = (price0Data.data as TokenReturns).concat(...(price0Data3M.data as TokenReturns));
+          const prices1 = (price1Data.data as TokenReturns).concat(...(price1Data3M.data as TokenReturns));
+
+          const twentyFourHoursAgo = Date.now() / 1000 - 24 * 60 * 60;
+
+          let cashBasis = 0;
+          let cashBasis1D = 0;
+          let shareDiff1D = 0;
+          
+          for (const transfer of transfers) {
+            const addrFrom: string = transfer.topics[1].slice(26).toLowerCase();
+            const addrTo: string = transfer.topics[2].slice(26).toLowerCase();
+            const amountShares = toBig(BigNumber.from(transfer.data)).div('1e18');
+
+            if (addrTo === addrFrom) continue;
+
+            // Search for the nearest pricing data (by timestamp)
+            const R = findNearestElementByTime(poolReturns, transfer.timeStamp);
+            const P0 = findNearestElementByTime(prices0, transfer.timeStamp);
+            const P1 = findNearestElementByTime(prices1, transfer.timeStamp);
+            
+            const valueToken0 = amountShares.mul(R.inventory0).div(R.total_supply).mul(P0.price);
+            const valueToken1 = amountShares.mul(R.inventory1).div(R.total_supply).mul(P1.price);
+            const valueAll = valueToken0.toNumber() + valueToken1.toNumber();
+
+            const isWithin1D = Number(transfer.timeStamp) > twentyFourHoursAgo;
+            if (addrTo === accountAddress) {
+              cashBasis += valueAll;
+              if (isWithin1D) {
+                cashBasis1D += valueAll;
+                shareDiff1D += amountShares.toNumber();
+              };
+            } else {
+              cashBasis -= valueAll;
+              if (isWithin1D) {
+                cashBasis1D -= valueAll;
+                shareDiff1D -= amountShares.toNumber();
+              };
+            }
+          }
+
+          let accountValue1DAgo: number | null = null;
+          {
+            const R = findNearestElementByTime(poolReturns, twentyFourHoursAgo);
```

Comment:
Couldn't we just calculate the nearest timestamp once and use it to get an element for each (assuming the timestamps line up). We don't have to do this now, but this is part of the "better" idea I had in mind, let's look into this later.

Reply:
I tried that first, but unfortunately the timestamps only lined up for the first 5 elements of the arrays I was getting